RealmModelGenerator/AttributeDetailView.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  AttributeDetailView.swift
+//  RealmModelGenerator
+//
+//  Created by Zhaolong Zhong on 3/29/16.
+//  Copyright © 2016 QuarkWorks. All rights reserved.
+//
+
+import Cocoa
+
+@objc protocol AttributeDetailViewDelegate {
+    optional func attributeDetailView(attributeDetailView:AttributeDetailView, shouldChangeAttributeName name:String) -> Bool
+}
+
+@IBDesignable
+class AttributeDetailView: NibDesignableView, NSTextFieldDelegate {
+    static let TAG = NSStringFromClass(AttributeDetailView)
+    
+    weak var delegate:AttributeDetailViewDelegate?
+    
+    @IBOutlet var nameTextField: NSTextField! {
+        willSet {
+            newValue!.delegate = self
```

Comment:
also this can be handled in the nib if you want to. wire the delegate to the files owner.